packages/features/bookings/components/AvailableTimes.tsx

```diff
@@ -7,24 +11,164 @@ import { useLocale } from "@calcom/lib/hooks/useLocale";
+}: {
+  slot: Slots[string][number];
+  seatsPerTimeSlot?: number | null;
+  selectedSlots?: string[];
+  onTimeSelect: TOnTimeSelect;
+  showAvailableSeatsCount?: boolean | null;
+}) => {
+  const { t } = useLocale();
+
+  const overlayCalendarToggled = getQueryParam("overlayCalendar") === "true";
+  const [timeFormat, timezone] = useTimePreferences((state) => [state.timeFormat, state.timezone]);
+  const selectedDuration = useBookerStore((state) => state.selectedDuration);
+  const bookingData = useBookerStore((state) => state.bookingData);
+  const layout = useBookerStore((state) => state.layout);
+  const hasTimeSlots = !!seatsPerTimeSlot;
+  const computedDateWithUsersTimezone = dayjs.utc(slot.time).tz(timezone);
+
+  const bookingFull = !!(hasTimeSlots && slot.attendees && slot.attendees >= seatsPerTimeSlot);
+  const isHalfFull = slot.attendees && seatsPerTimeSlot && slot.attendees / seatsPerTimeSlot >= 0.5;
+  const isNearlyFull = slot.attendees && seatsPerTimeSlot && slot.attendees / seatsPerTimeSlot >= 0.83;
+  const colorClass = isNearlyFull ? "bg-rose-600" : isHalfFull ? "bg-yellow-500" : "bg-emerald-400";
+
+  const { isOverlapping, overlappingTimeEnd, overlappingTimeStart } = useCheckOverlapWithOverlay(
+    computedDateWithUsersTimezone,
+    selectedDuration
+  );
+  const [overlapConfirm, setOverlapConfirm] = useState(false);
+
+  const onButtonClick = useCallback(() => {
+    if (!overlayCalendarToggled) {
+      onTimeSelect(slot.time, slot?.attendees || 0, seatsPerTimeSlot, slot.bookingUid);
+      return;
+    }
+    if (isOverlapping && overlapConfirm) {
+      setOverlapConfirm(false);
+      return;
+    }
+
+    if (isOverlapping && !overlapConfirm) {
+      setOverlapConfirm(true);
+      return;
+    }
+    if (!overlapConfirm) {
+      onTimeSelect(slot.time, slot?.attendees || 0, seatsPerTimeSlot, slot.bookingUid);
+    }
+  }, [
+    overlayCalendarToggled,
+    isOverlapping,
+    overlapConfirm,
+    onTimeSelect,
+    slot.time,
+    slot?.attendees,
+    slot.bookingUid,
+    seatsPerTimeSlot,
+  ]);
+
+  return (
+    <AnimatePresence>
+      <div className="flex gap-2">
+        <Button
+          key={slot.time}
+          disabled={bookingFull || !!(slot.bookingUid && slot.bookingUid === bookingData?.uid)}
+          data-testid="time"
+          data-disabled={bookingFull}
+          data-time={slot.time}
+          onClick={onButtonClick}
+          className={classNames(
+            "min-h-9 hover:border-brand-default mb-2 flex h-auto w-full flex-grow flex-col justify-center py-2",
+            selectedSlots?.includes(slot.time) && "border-brand-default"
+          )}
+          color="secondary">
+          <div className="flex items-center gap-2">
+            {!hasTimeSlots && overlayCalendarToggled && (
+              <span
+                className={classNames(
+                  "inline-block h-2 w-2 rounded-full",
+                  isOverlapping ? "bg-rose-600" : "bg-emerald-400"
```

Comment:
Nit: Dots are not exactly in line
![Screenshot 2023-10-10 at 10 03 03](https://github.com/calcom/cal.com/assets/30310907/08c9b751-4f83-40b7-9bfe-1eff0b91d42c)